src/DBConnectionCaster.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "DBConnection.hpp"
+
+template <class To>
+struct DBConnectionCaster {
+    static const To& cast(const IDBConnection& object) {
+        std::cout << "Dynamic cast" << std::endl;
```

Comment:
@Morheit , you cannot verify `cout` in automatically in UT. that's why I don't like when `cout` used to prove something.
I suggest to drop `cout` at all. Add UT which will prove that if you'll feed some other class then dynamic cast will throw.

Reply:
Also it would be nice to have test case when dynamic will work fine. E.g. some other  SQLiteDBConnection::query() ^ cast and call.

Reply:
@malirod , std::cout removed. UT added